crates/storage/nippy-jar/src/lib.rs

```diff
@@ -235,39 +235,61 @@ where
 
         // SAFETY: File is read-only and its descriptor is kept alive as long as the mmap handle.
         let data_reader = unsafe { memmap2::Mmap::map(&data_file)? };
-        let mut obj: Self = bincode::deserialize_from(data_reader.as_ref())?;
+        let max_row_size: [u8; 8] =
+            data_reader.as_ref()[0..8].try_into().expect("slice with incorrect length");
```

Comment:
nit: the error message is not obvious, should it rather be something about missing header?